modules/scripts/startup-script/templates/startup-script-custom.tpl

```diff
@@ -29,7 +29,7 @@ stdlib::runner() {
   case "$1" in
     ansible-local) stdlib::run_playbook "$destpath/$filename" "$args";;
     # shellcheck source=/dev/null
-    shell)  sh -c "source '$destpath/$filename' $args";;
+    shell)  sh -c ". '$destpath/$filename' $args";;
```

Comment:
Here, on line 13, I would add a set -x so it fails when the runner fails. 
Also, I recommend replacing line 13 by: 
  . '$destpath/$filename' $args"
  exit_code=$?
  stdlib::debug "=== $object finished with exit_code=$exit_code ==="

Reply:
Done